<!DOCTYPE html><html lang="en-us" >


<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  
  
  
    <meta name="generator" content="Wowchemy 5.6.0 for Hugo" />
  

  
  












  
  










  







  
  

  
  
  

  
  
  
  
  
  

  

  
  
  
    
  
  <meta name="description" content="The Gene Ontology (GO) knowledgebase is the world’s largest source of information on the functions of genes. This knowledge is both human-readable and machine-readable, and is a foundation for computational analysis of large-scale molecular biology and genetics experiments in biomedical research." />

  
  <link rel="alternate" hreflang="en-us" href="https://example.com/tool/2019-06-30-gene-annot-anal/" />

  
  
  
    <meta name="theme-color" content="#1565c0" />
  

  
  
    
    <script src="/js/mathjax-config.js"></script>
  

  

  <link rel="stylesheet" href="/css/vendor-bundle.min.c7b8d9abd591ba2253ea42747e3ac3f5.css" media="print" onload="this.media='all'">

  
  
  
    
    

    
    
    
    
      
      
    
    
    

    
    
    

    

    
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js" integrity="" crossorigin="anonymous" async></script>
      
    
      
      

      
      

      
    
      
      

      
      

      
    
  

  
  
  
  
  
  
  <link rel="stylesheet" href="/css/wowchemy.7579f2ea081d74b9cb40a7d984071feb.css" />

  
  
  
  
  
  
  
    
    
    <link rel="stylesheet" href="/css/libs/chroma/github-light.min.css" title="hl-light" media="print" onload="this.media='all'" >
    <link rel="stylesheet" href="/css/libs/chroma/dracula.min.css" title="hl-dark" media="print" onload="this.media='all'" disabled>
  

  
  



  


  


  




  
  
  

  
  

  
  
    <link rel="manifest" href="/manifest.webmanifest" />
  

  
  <link rel="icon" type="image/png" href="/media/icon_hu1efa9af2edf414d45fb59503b457ea16_13437_32x32_fill_lanczos_center_3.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hu1efa9af2edf414d45fb59503b457ea16_13437_180x180_fill_lanczos_center_3.png" />

  <link rel="canonical" href="https://example.com/tool/2019-06-30-gene-annot-anal/" />

  
  
  
  
  
  
  
  
    
    
  
  

  
  
    
    
  
  <meta property="twitter:card" content="summary" />
  
    <meta property="twitter:site" content="@wowchemy" />
    <meta property="twitter:creator" content="@wowchemy" />
  
  <meta property="og:site_name" content="BM Sci-Tech Memento" />
  <meta property="og:url" content="https://example.com/tool/2019-06-30-gene-annot-anal/" />
  <meta property="og:title" content="Gene annotation and analysis resources | BM Sci-Tech Memento" />
  <meta property="og:description" content="The Gene Ontology (GO) knowledgebase is the world’s largest source of information on the functions of genes. This knowledge is both human-readable and machine-readable, and is a foundation for computational analysis of large-scale molecular biology and genetics experiments in biomedical research." /><meta property="og:image" content="https://example.com/media/icon_hu1efa9af2edf414d45fb59503b457ea16_13437_512x512_fill_lanczos_center_3.png" />
    <meta property="twitter:image" content="https://example.com/media/icon_hu1efa9af2edf414d45fb59503b457ea16_13437_512x512_fill_lanczos_center_3.png" /><meta property="og:locale" content="en-us" />
  
    
      <meta
        property="article:published_time"
        content="2019-06-30T00:00:00&#43;00:00"
      />
    
    <meta property="article:modified_time" content="2022-12-15T00:00:00&#43;00:00">
  

  



  

  


  <title>Gene annotation and analysis resources | BM Sci-Tech Memento</title>

  
  
  
  











</head>


<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="page-wrapper   " data-wc-page-id="e673048ec594f437a54d3b648c74ee4e" >

  
  
  
  
  
  
  
  
  
  <script src="/js/wowchemy-init.min.613040fe4f2c0f007b4dcb64404201cb.js"></script>

  


<aside class="search-modal" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#" aria-label="Close"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search" class="form-control"
        aria-label="Search...">
        
      </div>

      
      

      

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>



  <div class="page-header">
    












<header class="header--fixed">
  <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
    <div class="container-xl">

      
      <div class="d-none d-lg-inline-flex">
        <a class="navbar-brand" href="/">BM Sci-Tech Memento</a>
      </div>
      

      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar-content" aria-controls="navbar-content" aria-expanded="false" aria-label="Toggle navigation">
      <span><i class="fas fa-bars"></i></span>
      </button>
      

      
      <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
        <a class="navbar-brand" href="/">BM Sci-Tech Memento</a>
      </div>
      

      
      
      <div class="navbar-collapse main-menu-item collapse justify-content-end" id="navbar-content">

        
        <ul class="navbar-nav d-md-inline-flex">
          

          

          
          
          

          

          
          
          
          

          
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/"><span>Home</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link  active" href="/tool/"><span>Tools</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/docs/"><span>Notes</span></a>
          </li>

          
          

          

          
          
          
            
              
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="https://pubmed.ncbi.nlm.nih.gov/" target="_blank" rel="noopener"><span>PubMed</span></a>
          </li>

          
          

          
          <li class="nav-item dropdown">
            <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-haspopup="true"><span>NCBI</span><span class="caret"></span>
            </a>
            <div class="dropdown-menu">
              
                <a class="dropdown-item" href="https://www.ncbi.nlm.nih.gov/gene/"><span>Gene</span></a>
              
                <a class="dropdown-item" href="https://www.ncbi.nlm.nih.gov/nucleotide/"><span>Nucleotide</span></a>
              
                <a class="dropdown-item" href="https://www.ncbi.nlm.nih.gov/geo/"><span>GEO</span></a>
              
                <a class="dropdown-item" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi"><span>BLAST</span></a>
              
            </div>
          </li>

          
          

          

          
          
          
            
              
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="https://licore.netlify.app/" target="_blank" rel="noopener"><span>MII</span></a>
          </li>

          
          

        

          
        </ul>
      </div>

      <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">

        
        

        
        
        <li class="nav-item">
          <a class="nav-link js-search" href="#" aria-label="Search"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        
        
        
        <li class="nav-item dropdown theme-dropdown">
          <a href="#" class="nav-link" data-toggle="dropdown" aria-haspopup="true" aria-label="Display preferences">
            <i class="fas fa-moon" aria-hidden="true"></i>
          </a>
          <div class="dropdown-menu">
            <a href="#" class="dropdown-item js-set-theme-light">
              <span>Light</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-dark">
              <span>Dark</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-auto">
              <span>Automatic</span>
            </a>
          </div>
        </li>
        

        
        

      </ul>

    </div>
  </nav>
</header>


  </div>

  <div class="page-body">
    
    
    

    <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>Gene annotation and analysis resources</h1>

  

  
    


<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
          Last updated on
      
    
    Thursday, 15 December 2022 
  </span>
  

  

  

  
  
  
  

  
  

</div>

    





  
</div>



  <div class="article-container">

    <div class="article-style">
      <p>The <a href="http://geneontology.org/" target="_blank" rel="noopener"><strong><em>Gene Ontology (GO)</em></strong></a> knowledgebase is the world’s largest source of information on the functions of genes. This knowledge is both human-readable and machine-readable, and is a foundation for computational analysis of large-scale molecular biology and genetics experiments in biomedical research. The mission of the <a href="http://geneontology.org/" target="_blank" rel="noopener"><strong><em>GO</em></strong></a> Consortium is to develop a comprehensive, computational model of biological systems, ranging from the molecular to the organism level, across the multiplicity of species in the tree of life.</p>
<p><a href="https://www.genome.jp/kegg/" target="_blank" rel="noopener"><strong><em>Kyoto Encyclopedia of Genes and Genomes (KEGG)</em></strong></a> is a database resource for understanding high-level functions and utilities of the biological system, such as the cell, the organism and the ecosystem, from molecular-level information, especially large-scale molecular datasets generated by genome sequencing and other high-throughput experimental technologies.</p>
<p><a href="https://reactome.org/" target="_blank" rel="noopener"><strong><em>REACTOME</em></strong></a> is an open-source, open access, manually curated and peer-reviewed pathway database. The goal is to provide intuitive bioinformatics tools for the visualization, interpretation and analysis of pathway knowledge to support basic and clinical research, genome analysis, modeling, systems biology and education. Founded in 2003, the Reactome project is led by Lincoln Stein of <a href="https://oicr.on.ca/" target="_blank" rel="noopener">OICR</a>, Peter D’Eustachio of <a href="https://nyulangone.org/" target="_blank" rel="noopener">NYULMC</a>, Henning Hermjakob of <a href="https://www.ebi.ac.uk/" target="_blank" rel="noopener">EMBL-EBI</a>, and Guanming Wu of <a href="https://www.ohsu.edu/" target="_blank" rel="noopener">OHSU</a>. <a href="https://reactome.org/" target="_blank" rel="noopener"><strong><em>REACTOME</em></strong></a> is part of the <a href="https://www.elixir-europe.org/platforms/data/core-data-resources" target="_blank" rel="noopener">ELIXIR Core Data Resources</a>. You can click to see more about <a href="https://www.elixir-europe.org/" target="_blank" rel="noopener">ELIXIR</a>.</p>
<p><a href="https://www.gsea-msigdb.org/" target="_blank" rel="noopener"><em><strong>Gene Set Enrichment Analysis (GSEA)</strong></em></a> is a computational method that determines whether an a priori defined set of genes shows statistically significant, concordant differences between two biological states.</p>
<p><a href="https://metascape.org/gp/index.html" target="_blank" rel="noopener"><strong><em>Metascape</em></strong></a> provides a rather unique protein-protein interaction (PPI) network analysis capability. In many gene list analysis resources, PPI analysis results in a rather massy hairball network. Besides stating such networks are statistically significant, there is not much biologists can say about such networks. To infer more biologically interpretable results, <a href="https://metascape.org/gp/index.html" target="_blank" rel="noopener"><strong><em>Metascape</em></strong></a> applies a mature complex identification algorithm called <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-4-2" target="_blank" rel="noopener">MCODE</a> to automatically extract protein complexes embedded in such large network. Then taking advantage of <a href="https://metascape.org/gp/index.html" target="_blank" rel="noopener"><strong><em>Metascape</em></strong></a>’s functional enrichment analysis capability, it automatically assigns putative biological roles of each <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-4-2" target="_blank" rel="noopener">MCODE</a> complex. Such analyses are very computational intensive and cannot be easily computed even by bioinformaticians. Regardless of its advanced PPI analysis algorithm, the results still heavily determined by the quality of its underlying PPI database.</p>
<p>The <strong>D</strong>atabase for <strong>A</strong>nnotation, <strong>V</strong>isualization and <strong>I</strong>ntegrated <strong>D</strong>iscovery (<a href="https://david.ncifcrf.gov/home.jsp" target="_blank" rel="noopener"><strong><em>DAVID</em></strong></a>) provides a comprehensive set of functional annotation tools for investigators to understand the biological meaning behind large lists of genes. These tools are powered by the comprehensive <a href="https://david.ncifcrf.gov/home.jsp" target="_blank" rel="noopener"><strong><em>DAVID</em></strong></a> Knowledgebase built upon the <a href="https://david.ncifcrf.gov/home.jsp" target="_blank" rel="noopener"><strong><em>DAVID</em></strong></a> Gene concept which pulls together multiple sources of functional annotations. For any given gene list, <a href="https://david.ncifcrf.gov/home.jsp" target="_blank" rel="noopener"><strong><em>DAVID</em></strong></a> tools are able to:</p>
<ul>
<li>Identify enriched biological themes, particularly GO terms</li>
<li>Discover enriched functional-related gene groups</li>
<li>Cluster redundant annotation terms</li>
<li>Visualize genes on BioCarta &amp; KEGG pathway maps</li>
<li>Display related many-genes-to-many-terms on 2-D view</li>
<li>Search for other functionally related genes not in the list</li>
<li>List interacting proteins</li>
<li>Explore gene names in batch</li>
<li>Link gene-disease associations</li>
<li>Highlight protein functional domains and motifs</li>
<li>Redirect to related literatures</li>
<li>Convert gene identifiers from one type to another</li>
</ul>
<p>Profiling samples from patients, tissues, and cells with genomics, transcriptomics, epigenomics, proteomics, and metabolomics ultimately produces lists of genes and proteins that need to be further analyzed and integrated in the context of known biology. <a href="https://maayanlab.cloud/Enrichr/" target="_blank" rel="noopener"><strong><em>Enrichr</em></strong></a> is a gene set search engine that enables the querying of hundreds of thousands of annotated gene sets. <a href="https://maayanlab.cloud/Enrichr/" target="_blank" rel="noopener"><strong><em>Enrichr</em></strong></a> uniquely integrates knowledge from many high-profile projects to provide synthesized information about mammalian genes and gene sets. The platform provides various methods to compute gene set enrichment, and the results are visualized in several interactive ways. This protocol provides a summary of the key features of <a href="https://maayanlab.cloud/Enrichr/" target="_blank" rel="noopener"><strong><em>Enrichr</em></strong></a>, which include using <a href="https://maayanlab.cloud/Enrichr/" target="_blank" rel="noopener"><strong><em>Enrichr</em></strong></a> programmatically and embedding an <a href="https://maayanlab.cloud/Enrichr/" target="_blank" rel="noopener"><strong><em>Enrichr</em></strong></a> button on any website. To learn more about the platform, <a href="https://pubmed.ncbi.nlm.nih.gov/33780170/" target="_blank" rel="noopener">click here</a> to access the free articles on PubMed for protocols.</p>
<p><a href="https://cn.string-db.org/" target="_blank" rel="noopener"><strong><em>STRING</em></strong></a> is part of the <a href="https://www.elixir-europe.org/platforms/data/core-data-resources" target="_blank" rel="noopener">ELIXIR Core Data Resources</a>. You can click to see more about <a href="https://www.elixir-europe.org/" target="_blank" rel="noopener">ELIXIR</a>.</p>
<p><a href="https://www.genecards.org/" target="_blank" rel="noopener"><strong><em>GeneCards</em></strong></a> is a searchable, integrative database that provides comprehensive, user-friendly information on all annotated and predicted human genes. The knowledgebase automatically integrates gene-centric data from about 150 web sources, including genomic, transcriptomic, proteomic, genetic, clinical and functional information.</p>
<p><a href="genemania.org"><strong><em>GeneMANIA</em></strong></a> helps you predict the function of your favourite genes and gene sets. <a href="genemania.org"><strong><em>GeneMANIA</em></strong></a> finds other genes that are related to a set of input genes, using a very large set of functional association data. Association data include protein and genetic interactions, pathways, co-expression, co-localization and protein domain similarity. You can use <a href="genemania.org"><strong><em>GeneMANIA</em></strong></a> to find new members of a pathway or complex, find additional genes you may have missed in your screen or find new genes with a specific function, such as protein kinases. Your question is defined by the set of genes you input.</p>
<p><a href="http://www.funrich.org/" target="_blank" rel="noopener"><strong><em>Functional Enrichment analysis tool (FunRich)</em></strong></a> is a stand-alone software tool used mainly for functional enrichment and interaction network analysis of genes and proteins. Besides, the results of the analysis can be depicted graphically in the form of Venn, Bar, Column, Pie and Doughnut charts. Currently, <a href="http://www.funrich.org/" target="_blank" rel="noopener"><strong><em>FunRich</em></strong></a> tool is designed to handle variety of gene/protein data sets irrespective of the organism. Users can not only search against default background database, but can also load customized database against which functional enrichment analysis can be carried out.</p>
<p><a href="http://www.webgestalt.org/" target="_blank" rel="noopener"><strong>WebGestalt</strong></a> (WEB-based Gene SeT AnaLysis Toolkit) is a functional enrichment analysis web tool, which has on average 26,000 unique users from 144 countries and territories per year according to Google Analytics. <a href="http://www.webgestalt.org/" target="_blank" rel="noopener"><strong>WebGestalt</strong></a> 2019 significantly improved the output report with emphasis on providing user-friendly interfaces which could directly translate into publication-ready figures. The R package <a href="https://cran.r-project.org/web/packages/WebGestaltR/index.html" target="_blank" rel="noopener">WebGestaltR</a> has been updated to work with the new verison, which provides an interface to integrate into other pipelines or run batch jobs locally. <a href="http://www.webgestalt.org/" target="_blank" rel="noopener"><strong>WebGestalt</strong></a> also supports loading data from third-party websites or services through an API to perform enrichment analysis. <a href="http://www.webgestalt.org/" target="_blank" rel="noopener"><strong>WebGestalt</strong></a> supports three well-established and complementary methods for enrichment analysis, including Over-Representation Analysis (ORA), Gene Set Enrichment Analysis (GSEA), and Network Topology-based Analysis (NTA).</p>
<p>Biological data analysis often deals with lists of genes arising from various studies. The <a href="https://biit.cs.ut.ee/gprofiler/gost" target="_blank" rel="noopener"><strong>g:Profiler</strong></a> toolset is widely used for finding biological categories enriched in gene lists, conversions between gene identifiers and mappings to their orthologs. The mission of <a href="https://biit.cs.ut.ee/gprofiler/gost" target="_blank" rel="noopener"><strong>g:Profiler</strong></a> is to provide a reliable service based on up-to-date high quality data in a convenient manner across many evidence types, identifier spaces and organisms. <a href="https://biit.cs.ut.ee/gprofiler/gost" target="_blank" rel="noopener"><strong>g:Profiler</strong></a> relies on Ensembl as a primary data source and follows their quarterly release cycle while updating the other data sources simultaneously. The current update provides a better user experience due to a modern responsive web interface, standardised API and libraries. The results are delivered through an interactive and configurable web design. Results can be downloaded as publication ready visualisations or delimited text files. In the current update we have extended the support to 467 species and strains, including vertebrates, plants, fungi, insects and parasites. By supporting user uploaded custom GMT files, <a href="https://biit.cs.ut.ee/gprofiler/gost" target="_blank" rel="noopener"><strong>g:Profiler</strong></a> is now capable of analysing data from any organism. All past releases are maintained for reproducibility and transparency. The 2019 update introduces an extensive technical rewrite making the services faster and more flexible.</p>
<p><a href="https://cibersortx.stanford.edu/" target="_blank" rel="noopener"><strong><em>CIBERSORTx</em></strong></a> is an analytical tool from the <a href="https://med.stanford.edu/ash_alizadeh.html" target="_blank" rel="noopener">Alizadeh Lab</a> and <a href="https://anlab.stanford.edu/" target="_blank" rel="noopener">Newman Lab</a> to impute gene expression profiles and provide an estimation of the abundances of member cell types in a mixed cell population, using gene expression data.</p>
<p><a href="https://maayanlab.cloud/archs4/" target="_blank" rel="noopener"><strong><em>ARCHS4</em></strong></a> provides access to gene counts from HiSeq 2000, HiSeq 2500 and NextSeq 500 platforms for human and mouse experiments from <a href="https://www.ncbi.nlm.nih.gov/geo/" target="_blank" rel="noopener">GEO</a> and <a href="https://www.ncbi.nlm.nih.gov/sra" target="_blank" rel="noopener">SRA</a>. The website enables downloading of the data in H5 format for programmatic access as well as a 3-dimensional view of the sample and gene spaces. Search features allow browsing of the data by meta data annotation, ability to submit your own up and down gene sets, and explore matching samples enriched for annotated gene sets. Selected sample sets can be downloaded into a tab separated text file through auto-generated <a href="https://www.r-project.org/" target="_blank" rel="noopener">R</a> scripts for further analysis. Reads are aligned with Kallisto using a custom cloud computing platform. The artice about <a href="https://maayanlab.cloud/archs4/" target="_blank" rel="noopener"><strong><em>ARCHS4</em></strong></a> was published on <a href="https://www.nature.com/articles/s41467-018-03751-6" target="_blank" rel="noopener">Nature Communications</a> and can be found on <a href="https://pubmed.ncbi.nlm.nih.gov/29636450/" target="_blank" rel="noopener">PubMed</a> as well.</p>
<p><a href="https://lisanwanglab.org/SPAR" target="_blank" rel="noopener"><strong><em>SPAR</em></strong></a> integrates significantly improved mapping, segmentation, annotation, and RNA processing information for both sncRNA genes and mature sncRNA products. The improved mapping by <a href="https://lisanwanglab.org/SPAR" target="_blank" rel="noopener"><strong><em>SPAR</em></strong></a> provides significantly more data for downstream analysis (on average, over 28% more reads per library). <a href="https://lisanwanglab.org/SPAR" target="_blank" rel="noopener"><strong><em>SPAR</em></strong></a> segmentation is not only signficantly more accurate, but also is order-of-magnitude faster (60x speed-up on average) by utilizing novel segmentation algorithms. <a href="https://lisanwanglab.org/SPAR" target="_blank" rel="noopener"><strong><em>SPAR</em></strong></a> has been used to process over 800 smRNA high-throughput sequencing (smRNA-seq) datasets from <a href="https://www.ncbi.nlm.nih.gov/geo/" target="_blank" rel="noopener">GEO</a>, <a href="https://www.ncbi.nlm.nih.gov/sra" target="_blank" rel="noopener">SRA</a> and <a href="https://www.encodeproject.org/" target="_blank" rel="noopener">ENCODE</a> with over 20 billion reads. <a href="https://lisanwanglab.org/SPAR" target="_blank" rel="noopener"><strong><em>SPAR</em></strong></a> results are integrated into the <a href="http://lisanwanglab.org/DASHR" target="_blank" rel="noopener">database of small human non-coding RNAs (DASHR)</a> described in NAR 2016 Database issue.</p>
<ul class="cta-group">
  
  <li>
    <a href="/tag/databases/"  class="btn btn-primary px-3 py-3"><font color='white'> &lt; Back to Databases</font></a>
  </li>
  
  
  <li>
    <a href="/tool/" >
      Alternative to Tools <i class="fas fa-angle-right"></i>
    </a>
  </li>
  
</ul>


    </div>

    





<div class="article-tags">
  
  <a class="badge badge-light" href="/tag/databases/">Databases</a>
  
  <a class="badge badge-light" href="/tag/bioinformatics/">Bioinformatics</a>
  
</div>



<div class="share-box">
  <ul class="share">
    
  </ul>
</div>


























  </div>
</article>
  </div>

  <div class="page-footer">
    
    
    <div class="container">
      <footer class="site-footer">

  












  

  

  

  
  






  
  <p class="powered-by copyright-license-text">
    Copyright © 2023 | Biomedist | All rights reserved.
  </p>
  




  <p class="powered-by">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://wowchemy.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Wowchemy</a> — the free, <a href="https://github.com/wowchemy/wowchemy-hugo-themes" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>
</footer>

    </div>
    
  </div>

  


<script src="/js/vendor-bundle.min.d26509351aa0ff874abbee824e982e9b.js"></script>




  

  
  

  













  
  <script id="search-hit-fuse-template" type="text/x-template">
    <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
    </div>
  </script>
  
    <script src="https://cdn.jsdelivr.net/gh/krisk/Fuse@v3.2.1/dist/fuse.min.js" integrity="sha512-o38bmzBGX+hD3JHWUFCDA09btWaqrNmoJ3RXLlrysA7PP01Kgs4UlE4MhelE1v5dJR3+cxlR4qQlotsW7jKsnw==" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/gh/julmot/mark.js@8.11.1/dist/jquery.mark.min.js" integrity="sha512-mhbv5DqBMgrWL+32MmsDOt/OAvqr/cHimk6B8y/bx/xS88MVkYGPiVv2ixKVrkywF2qHplNRUvFsAHUdxZ3Krg==" crossorigin="anonymous"></script>
  












  
  
  
  
  
  
  







<script id="page-data" type="application/json">{"use_headroom":true}</script>



  <script src="/js/wowchemy-headroom.c251366b4128fd5e6b046d4c97a62a51.js" type="module"></script>








  
  


<script src="/en/js/wowchemy.min.54dd6e4d8f2e4b1d098381b57f18dd83.js"></script>























</body>
</html>
