<!DOCTYPE html><html lang="en-us" >


<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  
  
  
    <meta name="generator" content="Wowchemy 5.6.0 for Hugo" />
  

  
  












  
  










  







  
  

  
  
  

  
  
  
  
  
  

  

  
  
  
    
  
  <meta name="description" content="Promoter2.0 predicts transcription start sites of vertebrate PolII promoters in DNA sequences. It has been developed as an evolution of simulated transcription factors that interact with sequences in promoter regions." />

  
  <link rel="alternate" hreflang="en-us" href="https://example.com/tool/2021-08-06-en-tf/" />

  
  
  
    <meta name="theme-color" content="#1565c0" />
  

  
  
    
    <script src="/js/mathjax-config.js"></script>
  

  

  <link rel="stylesheet" href="/css/vendor-bundle.min.c7b8d9abd591ba2253ea42747e3ac3f5.css" media="print" onload="this.media='all'">

  
  
  
    
    

    
    
    
    
      
      
    
    
    

    
    
    

    

    
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
    
      
      

      
      

      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js" integrity="" crossorigin="anonymous" async></script>
      
    
      
      

      
      

      
    
      
      

      
      

      
    
  

  
  
  
  
  
  
  <link rel="stylesheet" href="/css/wowchemy.7579f2ea081d74b9cb40a7d984071feb.css" />

  
  
  
  
  
  
  
    
    
    <link rel="stylesheet" href="/css/libs/chroma/github-light.min.css" title="hl-light" media="print" onload="this.media='all'" >
    <link rel="stylesheet" href="/css/libs/chroma/dracula.min.css" title="hl-dark" media="print" onload="this.media='all'" disabled>
  

  
  



  


  


  




  
  
  

  
  

  
  
    <link rel="manifest" href="/manifest.webmanifest" />
  

  
  <link rel="icon" type="image/png" href="/media/icon_hu1efa9af2edf414d45fb59503b457ea16_13437_32x32_fill_lanczos_center_3.png" />
  <link rel="apple-touch-icon" type="image/png" href="/media/icon_hu1efa9af2edf414d45fb59503b457ea16_13437_180x180_fill_lanczos_center_3.png" />

  <link rel="canonical" href="https://example.com/tool/2021-08-06-en-tf/" />

  
  
  
  
  
  
  
  
    
    
  
  

  
  
    
    
  
  <meta property="twitter:card" content="summary" />
  
    <meta property="twitter:site" content="@wowchemy" />
    <meta property="twitter:creator" content="@wowchemy" />
  
  <meta property="og:site_name" content="BM Sci-Tech Memento" />
  <meta property="og:url" content="https://example.com/tool/2021-08-06-en-tf/" />
  <meta property="og:title" content="Encyclopedia of transcription factor | BM Sci-Tech Memento" />
  <meta property="og:description" content="Promoter2.0 predicts transcription start sites of vertebrate PolII promoters in DNA sequences. It has been developed as an evolution of simulated transcription factors that interact with sequences in promoter regions." /><meta property="og:image" content="https://example.com/media/icon_hu1efa9af2edf414d45fb59503b457ea16_13437_512x512_fill_lanczos_center_3.png" />
    <meta property="twitter:image" content="https://example.com/media/icon_hu1efa9af2edf414d45fb59503b457ea16_13437_512x512_fill_lanczos_center_3.png" /><meta property="og:locale" content="en-us" />
  
    
      <meta
        property="article:published_time"
        content="2021-08-06T00:00:00&#43;00:00"
      />
    
    <meta property="article:modified_time" content="2022-12-30T00:00:00&#43;00:00">
  

  



  

  


  <title>Encyclopedia of transcription factor | BM Sci-Tech Memento</title>

  
  
  
  











</head>


<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" class="page-wrapper   " data-wc-page-id="353168d35d5e59f42fb223e30a636c14" >

  
  
  
  
  
  
  
  
  
  <script src="/js/wowchemy-init.min.613040fe4f2c0f007b4dcb64404201cb.js"></script>

  


<aside class="search-modal" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#" aria-label="Close"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search" class="form-control"
        aria-label="Search...">
        
      </div>

      
      

      

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>



  <div class="page-header">
    












<header class="header--fixed">
  <nav class="navbar navbar-expand-lg navbar-light compensate-for-scrollbar" id="navbar-main">
    <div class="container-xl">

      
      <div class="d-none d-lg-inline-flex">
        <a class="navbar-brand" href="/">BM Sci-Tech Memento</a>
      </div>
      

      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar-content" aria-controls="navbar-content" aria-expanded="false" aria-label="Toggle navigation">
      <span><i class="fas fa-bars"></i></span>
      </button>
      

      
      <div class="navbar-brand-mobile-wrapper d-inline-flex d-lg-none">
        <a class="navbar-brand" href="/">BM Sci-Tech Memento</a>
      </div>
      

      
      
      <div class="navbar-collapse main-menu-item collapse justify-content-end" id="navbar-content">

        
        <ul class="navbar-nav d-md-inline-flex">
          

          

          
          
          

          

          
          
          
          

          
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/"><span>Home</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link  active" href="/tool/"><span>Tools</span></a>
          </li>

          
          

          

          
          
          
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="/docs/"><span>Notes</span></a>
          </li>

          
          

          

          
          
          
            
              
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="https://pubmed.ncbi.nlm.nih.gov/" target="_blank" rel="noopener"><span>PubMed</span></a>
          </li>

          
          

          
          <li class="nav-item dropdown">
            <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-haspopup="true"><span>NCBI</span><span class="caret"></span>
            </a>
            <div class="dropdown-menu">
              
                <a class="dropdown-item" href="https://www.ncbi.nlm.nih.gov/gene/"><span>Gene</span></a>
              
                <a class="dropdown-item" href="https://www.ncbi.nlm.nih.gov/nucleotide/"><span>Nucleotide</span></a>
              
                <a class="dropdown-item" href="https://www.ncbi.nlm.nih.gov/geo/"><span>GEO</span></a>
              
                <a class="dropdown-item" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi"><span>BLAST</span></a>
              
            </div>
          </li>

          
          

          

          
          
          
            
              
            
          

          

          
          
          
          

          
            
              
              
            
            
          

          <li class="nav-item">
            <a class="nav-link " href="https://licore.netlify.app/" target="_blank" rel="noopener"><span>MII</span></a>
          </li>

          
          

        

          
        </ul>
      </div>

      <ul class="nav-icons navbar-nav flex-row ml-auto d-flex pl-md-2">

        
        

        
        
        <li class="nav-item">
          <a class="nav-link js-search" href="#" aria-label="Search"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        
        
        
        <li class="nav-item dropdown theme-dropdown">
          <a href="#" class="nav-link" data-toggle="dropdown" aria-haspopup="true" aria-label="Display preferences">
            <i class="fas fa-moon" aria-hidden="true"></i>
          </a>
          <div class="dropdown-menu">
            <a href="#" class="dropdown-item js-set-theme-light">
              <span>Light</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-dark">
              <span>Dark</span>
            </a>
            <a href="#" class="dropdown-item js-set-theme-auto">
              <span>Automatic</span>
            </a>
          </div>
        </li>
        

        
        

      </ul>

    </div>
  </nav>
</header>


  </div>

  <div class="page-body">
    
    
    

    <article class="article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1>Encyclopedia of transcription factor</h1>

  

  
    


<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
          Last updated on
      
    
    Friday, 30 December 2022 
  </span>
  

  

  

  
  
  
  

  
  

</div>

    





  
</div>



  <div class="article-container">

    <div class="article-style">
      <ul>
<li>
<p><a href="https://services.healthtech.dtu.dk/service.php?Promoter-2.0" target="_blank" rel="noopener"><strong>Promoter2.0</strong></a> predicts transcription start sites of vertebrate PolII promoters in DNA sequences. It has been developed as an evolution of simulated transcription factors that interact with sequences in promoter regions. i It builds on principles that are common to neural networks and genetic algorithms.</p>
</li>
<li>
<p><a href="https://jaspar.genereg.net/" target="_blank" rel="noopener"><strong>JASPAR</strong></a> is a regularly maintained open-access database storing manually curated transcription factors (TF) binding profiles as position frequency matrices (PFMs). PFMs summarize occurrences of each nucleotide at each position in a set of observed TF-DNA interactions. PFMs can be transformed to probabilistic or energistic models to construct position weight matrices (PWMs) or position-specific scoring matrices (PSSMs), which can be used to scan any DNA sequence to predict TF binding sites (TFBSs). The <a href="https://jaspar.genereg.net/" target="_blank" rel="noopener"><strong>JASPAR</strong></a> database provides TFBSs predicted using the profiles in the CORE collection.
The motifs in <a href="https://jaspar.genereg.net/" target="_blank" rel="noopener"><strong>JASPAR</strong></a> are collected in two ways:</p>
<ul>
<li>Internally: de novo generated motifs, by analyzing ChIP-seq/-exo sequences using a custom motif discovery pipeline (check the code at our repository).</li>
<li>Externally: motifs taken directly from other publications and/or resources.
In both cases, the selected motifs are manually curated. Specifically, our curators assess the quality of the motif and search for an orthogonal publication providing support to the motif as the bona fide motif recognized by the TF of interest (e.g., a motif found in ChIP-seq peaks looks similar to one found by SELEX-seq). The Pubmed ID associated with the orthogonal support is provided in the TF profile metadata.</li>
</ul>
</li>
</ul>
<blockquote>
<p><a href="https://jaspar.genereg.net/" target="_blank" rel="noopener"><strong>JASPAR</strong></a> is the only database with this scope where the data can be used with no restrictions (open source). For a comprehensive review of models and how they can be used, please see the following reviews:</p>
<ul>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/25045190/" target="_blank" rel="noopener"><em>Modeling the specificity of protein-DNA interactions Quant Biol. 2013 Jun;1(2):115-130</em></a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/15131651/" target="_blank" rel="noopener"><em>Applied bioinformatics for the identification of regulatory elements Nat Rev Genet. 2004 Apr;5(4):276-87</em></a></li>
</ul>
</blockquote>
<ul>
<li>
<p>The <a href="http://motifmap.ics.uci.edu/" target="_blank" rel="noopener"><strong>MotifMap</strong></a> system provides comprehensive maps of candidate regulatory elements encoded in the genomes of model species using databases of transcription factor binding motifs, refined genome alignments, and a comparative genomic statistical approach- Bayesian Branch Length Score. Users can filter the results by various criteria and export them to severals file formats or view them directly in the UCSC genome browser.</p>
</li>
<li>
<p><a href="http://gene-regulation.com/pub/databases.html" target="_blank" rel="noopener"><strong>TRANSFAC</strong></a> provides data on eukaryotic transcription factors, their experimentally-proven binding sites, consensus binding sequences (positional weight matrices) and regulated genes. TRANSCompel contains data on eukaryotic transcription factors experimentally proven to act together in a synergistic or antagonistic manner.</p>
</li>
<li>
<p>The <a href="http://the_brain.bwh.harvard.edu/uniprobe/index.php" target="_blank" rel="noopener"><strong>UniPROBE (Universal PBM Resource for Oligonucleotide Binding Evaluation)</strong></a> database hosts data generated by universal protein binding microarray (PBM) technology on the in vitro DNA binding specificities of proteins. This initial release of the <a href="http://the_brain.bwh.harvard.edu/uniprobe/index.php" target="_blank" rel="noopener"><strong>UniPROBE</strong></a> database provides a centralized resource for accessing comprehensive data on the preferences of proteins for all possible sequence variants (&lsquo;words&rsquo;) of length k (&lsquo;k-mers&rsquo;), as well as position weight matrix (PWM) and graphical sequence logo representations of the k-mer data. In total, the database currently hosts DNA binding data for 726 nonredundant proteins and complexes from a diverse collection of organisms, including the prokaryote Vibrio harveyi, the eukaryotic malarial parasite Plasmodium falciparum, the parasitic Apicomplexan Cryptosporidium parvum, the yeast Saccharomyces cerevisiae, the worm Caenorhabditis elegans, mouse, and human. The database&rsquo;s web tools (on the right) include a text-based search, a function for assessing motif similarity between user-entered data and database PWMs, and a function for locating putative binding sites along user-entered nucleotide sequences. Please click on each tool&rsquo;s &ldquo;help&rdquo; link for more information.</p>
</li>
<li>
<p>The <a href="http://bioinfo.wilmer.jhu.edu/PDI/" target="_blank" rel="noopener"><strong>Human Protein-DNA Interactome (hPDI)</strong></a> database holds experimental protein-DNA interaction data for humans identified by protein microarray assays. The current release of <a href="http://bioinfo.wilmer.jhu.edu/PDI/" target="_blank" rel="noopener"><strong>hPDI</strong></a> contains 17,718 protein-DNA interactions for 1013 human DNA-binding proteins. These DNA-binding proteins include 493 human transcription factors (TFs) and 520 unconventional DNA binding proteins (uDBPs). Here TFs are defined as proteins either are annotated as &ldquo;regulation of transcription&rdquo; defined by GO database or contain known DNA binding domains defined by Pfam database and, uDBPs are defined proteins that have not been previously found to binding DNA specifically.</p>
</li>
<li>
<p><a href="https://v1.factorbook.org/" target="_blank" rel="noopener"><strong>Factorbook</strong></a> is a transcription factor (TF)-centric web-based repository of integrative analysis associated with ENCODE ChIP-seq data. It includes de novo discovered motifs, chromatin features surrounding ChIP-seq peaks (histone modification patterns, DNase I cleavage footprints, and nucleosome positioning profiles), deep-learned models of sequence features driving TF binding, and integration with GWAS variants and the ENCODE Registry of candidate cis-regulatory elements.</p>
</li>
<li>
<p><a href="https://hocomoco.autosome.org/" target="_blank" rel="noopener"><strong>HOmo sapiens COmprehensive MOdel COllection (HOCOMOCO)</strong></a> v11 provides transcription factor (TF) binding models for 680 human and 453 mouse TFs. Since v11, <a href="https://hocomoco.autosome.org/" target="_blank" rel="noopener"><strong>HOCOMOCO</strong></a> is complemented by MoLoTool, an interactive web tool to mark motif occurrences in a given set of DNA sequences. In addition to basic mononucleotide position weight matrices (PWMs), <a href="https://hocomoco.autosome.org/" target="_blank" rel="noopener"><strong>HOCOMOCO</strong></a> provides dinucleotide position weight matrices based on ChIP-Seq data. All the models were produced by the ChIPMunk motif discovery tool. Model quality ratings are results of a comprehensive cross-validation benchmark. ChIP-Seq data for motif discovery was extracted from GTRD database of BioUML platform, that also provides an interface for motif finding (sequence scanning) with <a href="https://hocomoco.autosome.org/" target="_blank" rel="noopener"><strong>HOCOMOCO</strong></a> models.</p>
</li>
<li>
<p><a href="https://meme-suite.org/meme/index.html" target="_blank" rel="noopener"><strong>MEME</strong></a> discovers novel, ungapped motifs (recurring, fixed-length patterns) in your sequences (sample output from sequences). <a href="https://meme-suite.org/meme/index.html" target="_blank" rel="noopener"><strong>MEME</strong></a> splits variable-length patterns into two or more separate motifs. A motif is an approximate sequence pattern that occurs repeatedly in a group of related sequences. <a href="https://meme-suite.org/meme/index.html" target="_blank" rel="noopener"><strong>MEME</strong></a> represents motifs as position-dependent letter-probability matrices that describe the probability of each possible letter at each position in the pattern. Individual <a href="https://meme-suite.org/meme/index.html" target="_blank" rel="noopener"><strong>MEME</strong></a> motifs do not contain gaps. Patterns with variable-length gaps are split by <a href="https://meme-suite.org/meme/index.html" target="_blank" rel="noopener"><strong>MEME</strong></a> into two or more separate motifs. <a href="https://meme-suite.org/meme/index.html" target="_blank" rel="noopener"><strong>MEME</strong></a> takes as input a group of sequences and outputs as many motifs as requested. <a href="https://meme-suite.org/meme/index.html" target="_blank" rel="noopener"><strong>MEME</strong></a> uses statistical modeling techniques to automatically choose the best width, number of occurrences, and description for each motif. <a href="https://meme-suite.org/meme/index.html" target="_blank" rel="noopener"><strong>MEME</strong></a> on the web can take a second (control) set of input sequences and then discovers motifs that are enriched in the primary set relative to the control set. This is called discriminative motif discovery.</p>
</li>
<li>
<p><a href="http://cistrome.org/db/" target="_blank" rel="noopener"><strong>Cistrome DB</strong></a> uses the standard analysis pipeline ChiLin (1) to process all chromatin profiling reads.</p>
</li>
<li>
<p>Transcription factors (TFs) as key regulators could modulate the expression of target genes by binding to specific DNA sequences of their promoter(s) or enhancer(s). The spatiotemporal TF-target regulations play vital roles throughout the life course of organisms, especially for human beings. Thus, the identification of TF-target regulation in human is a basis for understanding the molecular regulatory mechanisms underlying biological processes including the development and pathogenesis. <a href="http://bioinfo.life.hust.edu.cn/hTFtarget#!/" target="_blank" rel="noopener"><strong>hTFtarget</strong></a> has curated comprehensive TF-target regulations from large-scale of ChIP-Seq data of human TFs (7,190 experiment samples of 659 TFs) in 569 conditions (399 types of cell line, 129 classes of tissues or cells, and 141 kinds of treatments). <a href="http://bioinfo.life.hust.edu.cn/hTFtarget#!/" target="_blank" rel="noopener"><strong>hTFtarget</strong></a> customized an analysis workflow to detect reliable TF-target regulations, which integrated the peak information of ChIP-Seq data with epigenomic modification status (Details in Document page). hTFtarget could help users to:</p>
<ul>
<li>Browse and investigate the target gene(s) of TF(s);</li>
<li>Explore which TF(s) could regulate the query gene(s);</li>
<li>Survey public ChIP-Seq samples for the query TF(s);</li>
<li>View ChIP-Seq peaks of the query TF(s) in user-defined manner;</li>
<li>Investigate potential co-associations between TFs in selected cell line(s);</li>
<li>Inquire candidate co-regulation of TFs for the query gene(s);</li>
<li>Predict binding sites of TF(s) on given sequence(s).</li>
</ul>
</li>
<li>
<p><a href="https://alggen.lsi.upc.es/" target="_blank" rel="noopener"><strong>PROMO</strong></a> is a virtual laboratory for the identification of putative transcription factor binding sites (TFBS) in DNA sequences from a species or groups of species of interest. TFBS defined in the TRANSFAC database are used to construct specific binding site weight matrices for TFBS prediction. The user can inspect the result of the search through a graphical interface and downloadable text files.</p>
</li>
<li>
<p><a href="www.softberry.com"><strong>Softberry</strong></a>, Inc. is a leading developer of software tools for genomic research focused on computational methods of high throughput biomedical data analysis, including software to support next generation sequencing technologies, transcriptome analysis (with RNASeq data), SNP detection and selection of disease specific SNP subsets. Its research-oriented programs have been used in thousand research projects. For example, fgenesh eukaryotic gene identification program is cited in more than 3000 articles, bprom, bacterial promoter prediction is cited in ~700 research papers. Just in 2015 our software applications have been used in more then 2000 research publications. Hundreds software applications developed in <a href="www.softberry.com"><strong>Softberry</strong></a> can be executed on-line at <a href="www.softberry.com"><strong>Softberry</strong></a> or downloaded for free academic usage. <a href="www.softberry.com"><strong>Softberry</strong></a> started its commercial activities on May 1997 and is headquartered in Mount Kisco, NY, USA, with a satellite office in Novosibirsk, Russia. It participated in a number of academic collaborations on various research projects around the world.</p>
</li>
<li>
<p><a href="http://promirna.molgen.mpg.de/" target="_blank" rel="noopener"><strong>PROmiRNA</strong></a> is a program for annotating miRNA promoters in human, as well as other species. It uses deepCAGE data from the FANTOM4 Consortium and integrated cage tag counts and other promoter features, such as CpG content, conservation and TATA box affinity, to score the potential of a candidate regions to be a promoter. Given a list of genomic regions of interest, in the form of a gff file, <a href="http://promirna.molgen.mpg.de/" target="_blank" rel="noopener"><strong>PROmiRNA</strong></a> returns the most probable promoter locations, together with the posterior probabilities calculated by the model. More informations on how to run <a href="http://promirna.molgen.mpg.de/" target="_blank" rel="noopener"><strong>PROmiRNA</strong></a> on your regions of interest can be found in the README file inside the <a href="http://promirna.molgen.mpg.de/" target="_blank" rel="noopener"><strong>PROmiRNA</strong></a> directory.</p>
</li>
</ul>
<ul class="cta-group">
  
  <li>
    <a href="/tag/databases/"  class="btn btn-primary px-3 py-3"><font color='white'> &lt; Back to Databases</font></a>
  </li>
  
  
  <li>
    <a href="/tool/" >
      Alternative to Tools <i class="fas fa-angle-right"></i>
    </a>
  </li>
  
</ul>


    </div>

    





<div class="article-tags">
  
  <a class="badge badge-light" href="/tag/databases/">Databases</a>
  
  <a class="badge badge-light" href="/tag/bioinformatics/">Bioinformatics</a>
  
  <a class="badge badge-light" href="/tag/data-mining/">Data Mining</a>
  
  <a class="badge badge-light" href="/tag/protein/">Protein</a>
  
  <a class="badge badge-light" href="/tag/interactome/">Interactome</a>
  
  <a class="badge badge-light" href="/tag/molecular-interactions/">Molecular Interactions</a>
  
  <a class="badge badge-light" href="/tag/dna/">DNA</a>
  
</div>



<div class="share-box">
  <ul class="share">
    
  </ul>
</div>


























  </div>
</article>
  </div>

  <div class="page-footer">
    
    
    <div class="container">
      <footer class="site-footer">

  












  

  

  

  
  






  
  <p class="powered-by copyright-license-text">
    Copyright © 2023 | Biomedist | All rights reserved.
  </p>
  




  <p class="powered-by">
    
    
    
      
      
      
      
      
      
      Published with <a href="https://wowchemy.com/?utm_campaign=poweredby" target="_blank" rel="noopener">Wowchemy</a> — the free, <a href="https://github.com/wowchemy/wowchemy-hugo-themes" target="_blank" rel="noopener">open source</a> website builder that empowers creators.
    
  </p>
</footer>

    </div>
    
  </div>

  


<script src="/js/vendor-bundle.min.d26509351aa0ff874abbee824e982e9b.js"></script>




  

  
  

  













  
  <script id="search-hit-fuse-template" type="text/x-template">
    <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
    </div>
  </script>
  
    <script src="https://cdn.jsdelivr.net/gh/krisk/Fuse@v3.2.1/dist/fuse.min.js" integrity="sha512-o38bmzBGX+hD3JHWUFCDA09btWaqrNmoJ3RXLlrysA7PP01Kgs4UlE4MhelE1v5dJR3+cxlR4qQlotsW7jKsnw==" crossorigin="anonymous"></script>
    <script src="https://cdn.jsdelivr.net/gh/julmot/mark.js@8.11.1/dist/jquery.mark.min.js" integrity="sha512-mhbv5DqBMgrWL+32MmsDOt/OAvqr/cHimk6B8y/bx/xS88MVkYGPiVv2ixKVrkywF2qHplNRUvFsAHUdxZ3Krg==" crossorigin="anonymous"></script>
  












  
  
  
  
  
  
  







<script id="page-data" type="application/json">{"use_headroom":true}</script>



  <script src="/js/wowchemy-headroom.c251366b4128fd5e6b046d4c97a62a51.js" type="module"></script>








  
  


<script src="/en/js/wowchemy.min.54dd6e4d8f2e4b1d098381b57f18dd83.js"></script>























</body>
</html>
